src/gsy_e/models/market/two_sided.py

```diff
@@ -344,15 +344,17 @@ def match_recommendations(
                     # re-raise exception to be handled by the external matcher
                     raise invalid_bop_exception
                 continue
-            original_bid_rate = market_bid.original_price / market_bid.energy
-            if ConstSettings.MASettings.BID_OFFER_MATCH_TYPE == BidOfferMatchAlgoEnum.PAY_AS_BID:
+            original_bid_rate = recommended_pair.bid_energy_rate + (
+                    market_bid.accumulated_grid_fees / recommended_pair.bid_energy)
+            if ConstSettings.MASettings.BID_OFFER_MATCH_TYPE == (
+                    BidOfferMatchAlgoEnum.PAY_AS_BID.value):
                 trade_rate = original_bid_rate
```

Comment:
I probably still not get it right, but I would expect:
```suggestion
                trade_rate = recommended_pair.bid_energy_rate + ( market_bid.accumulated_grid_fees / recommended_pair.bid_energy)
```

Reply:
You're right, this was messed up after several rounds of adjustments.
It is now fixed